src/OSPSuite.SimModelNative/include/SimModel/Species.h

```diff
@@ -36,10 +36,21 @@ class Species :
 
 	bool _negativeValuesAllowed;
 
+	Formula* createJacobianFor(const int index);
+	Formula* jacobianFor(const int index, const int mode, std::map<int, Formula*>* map);
```

Comment:
why not just `map<int, Formula*>& map` (reference to the map instead of pointer)